contracts/src/Gateway.sol

```diff
@@ -381,6 +389,20 @@ contract Gateway is IGateway, IInitializable, IUpgradable {
         emit PricingParametersChanged();
     }
 
+    // @dev Transact
+    function transact(bytes calldata data) external onlySelf {
+        TransactCallParams memory params = abi.decode(data, (TransactCallParams));
+        address agent = _ensureAgent(params.agentID);
+        bytes memory call =
+            abi.encodeCall(AgentExecutor.executeCall, (params.target, params.payload, params.dynamicGas));
+        (, bytes memory result) = Agent(payable(agent)).invoke(AGENT_EXECUTOR, call);
+        (bool success) = abi.decode(result, (bool));
+        if (!success) {
+            revert AgentTransactCallFailed();
+        }
+        emit TransactExecuted(params.agentID, params.target, params.payload);
```

Comment:
`params.payload` could be very large, cheaper to hash it when emitting event

```suggestion
        emit TransactExecuted(params.agentID, params.target, keccak256(params.payload));
```

Reply:
https://github.com/Snowfork/snowbridge/pull/1145/commits/7880d466ac4cf20b843279b041aa490b509f0483